### PR TITLE
Add hasspace

### LINKED ISSUE #15: has_space function
It would be nice to add `has_space` function. Currently, there is a `avx2_countspaces`, which counts all the number of spaces. I think it is possible to add another function to return `true` as soon as it finds space.

### PR DESCRIPTION
**Pull request by aminya**
Fixes #15

### CODE REVIEW

File: src/despacer.c [hunk 1]
```diff
@@ -1111,6 +1122,32 @@ size_t avx2_countspaces(const char *bytes, size_t howmany) {
...
+  __m256i carriage = _mm256_set1_epi8('\r');
+  size_t i = 0;
+  for (; i + 31 < howmany; i += 32) {
+    __m256i x = _mm256_loadu_si256((const __m256i *)(bytes + i));
+    // we do it the naive way, could be smarter?
+    __m256i xspaces = _mm256_cmpeq_epi8(x, spaces);
+    __m256i xnewline = _mm256_cmpeq_epi8(x, newline);
+    __m256i xcarriage = _mm256_cmpeq_epi8(x, carriage);
+    __m256i anywhite =
+        _mm256_or_si256(_mm256_or_si256(xspaces, xnewline), xcarriage);
```
Comment (lemire) [inline src/despacer.c:1137]: The identation of this code around here seems somewhat random. Code is easier to read when indentation follows a consistent model.

Comment (aminya) [reply src/despacer.c]: Fixed it.

**Comment by lemire** [comment]
Merging.

**Comment by aminya** [comment]
Thank you!

**Pull request merged**